src/main/java/com/google/cloud/genomics/utils/Paginator.java

```diff
@@ -369,6 +376,26 @@ public boolean apply(Read read) {
           .or(request));
     }
 
+    @Override
+    public GenomicsRequestInitializer<GenomicsRequest<?>> setFieldsInitializer(final String fields) {
+      return new GenomicsRequestInitializer<GenomicsRequest<?>>() {
+        @Override
+        public void initialize(GenomicsRequest<?> search) {
+          if (null != fields) {
+            for (String requiredField : REQUIRED_STRICT_SHARD_FIELDS.keySet()) {
```

Comment:
Perhaps this could be extracted to a common utility which accepts the fields string and the map of patterns to be matched.